test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -41,35 +41,20 @@ class test_uct_sockaddr : public uct_test {
 
     test_uct_sockaddr() : server(NULL), client(NULL), err_count(0),
                           server_recv_req(0), delay_conn_reply(false) {
-        memset(&listen_sock_addr,  0, sizeof(listen_sock_addr));
-        memset(&connect_sock_addr, 0, sizeof(connect_sock_addr));
     }
 
     void init() {
         uct_test::init();
 
         uct_iface_params_t server_params, client_params;
-        struct sockaddr_in *listen_addr_in, *connect_addr_in;
-
-        /* If we reached here, the interface is active, as it was tested at the
-         * resource creation */
-        if (!ucs::is_inet_addr((struct sockaddr *)&(GetParam()->connect_if_addr))) {
-            UCS_TEST_SKIP_R("There is no IP on the interface");
-        }
 
         /* This address is accessible, as it was tested at the resource creation */
-        listen_sock_addr.addr = (struct sockaddr *)&(GetParam()->listen_if_addr);
-        ASSERT_TRUE(listen_sock_addr.addr != NULL);
-
-        listen_addr_in = (struct sockaddr_in *) (listen_sock_addr.addr);
+        m_listen_addr  = GetParam()->listen_sock_addr;
+        m_connect_addr = GetParam()->connect_sock_addr;
 
-        /* Get a usable port on the host */
-        listen_addr_in->sin_port = ucs::get_port();
-
-        connect_sock_addr.addr = (struct sockaddr *)&(GetParam()->connect_if_addr);
-        ASSERT_TRUE(connect_sock_addr.addr != NULL);
-        connect_addr_in = (struct sockaddr_in *)connect_sock_addr.addr;
-        connect_addr_in->sin_port = listen_addr_in->sin_port;
+        uint16_t port = ucs::get_port();
```

Comment:
please declare the params in the begining of the scope. same for line 77

Reply:
is this actual for C++ code?

Reply:
I know it's common in c++ but seems more organized to me. IIRC we use this code convention in gtest and in the src files too